server/src/unifyfs_request_manager.c

```diff
@@ -2306,30 +2310,58 @@ static void chunk_read_response_rpc(hg_handle_t handle)
             hg_bulk_t bulk_handle;
             hret = margo_bulk_create(mid, 1, (void**)&resp_buf, &in.bulk_size,
                 HG_BULK_WRITE_ONLY, &bulk_handle);
-            assert(hret == HG_SUCCESS);
+            if (hret != HG_SUCCESS) {
+                LOGERR("failed to prepare bulk transfer");
```

Comment:
if we should still post a read response as noted below, this might also need to do the same.